Dockerfile

```diff
@@ -13,6 +13,15 @@ COPY LICENSE README.md THIRD_PARTY_NOTICE.md /
 
 RUN yarn global add @sanity/cli
 
+ARG NPM_SCOPE
+ARG NPM_TOKEN
+ARG REGISTRY_DOMAIN
+
+RUN if [[ -z "$NPM_SCOPE" && -z "$NPM_TOKEN" && -z "$REGISTRY_DOMAIN" ]]; \
+    then \
+    echo -e "//$REGISTRY_DOMAIN/:_authToken=$NPM_TOKEN\\n$NPM_SCOPE:registry=https://$REGISTRY_DOMAIN/" > .npmrc ; \
```

Comment:
A few issues with this:
1. If `.npmrc` exists then you just over-wrote it by using `>` instead of `>>`
2. If you use `>>` instead, and the existing file does not end with a line-break, then you've just invalidated the final line of the existing `.npmrc` by appending stuff to it.
3. This provides consumers with the ability to authorize exactly one registry and no more.

Regarding points 1 and 2, I think it's better to use `npm set \"//npm.pkg.github.com/:_authToken\" $GH_PACKAGE_REGISTRY_TOKEN --location=project`. It has the same net effect, but without making potentially destructive changes to any `.npmrc` files.

Not sure how to solve point 3 within this action, but do you even need to? Can you not just do this in the Workflow step prior to the `github-action-sanity` step? Eg.

```yml
name: Something

on:
  push:
    branches:
      - main

jobs:
  something:
    runs-on: ubuntu-latest
    steps:
      - name: Checkout
        uses: actions/checkout@v3
      - name: Authorize Package Registry
        run: npm set //foo.com/:_auth ${{ secrets.MY_TOKEN }}
      - name: Do Stuff
        uses: sanity-io/github-action-sanity
```

Reply:
Thanks for the review! Everything you said makes sense 👍🏼

If I get time to solutionize for 3, I'll update the PR. For my project I added a GH action step to use npm set like you suggested. However, I think it's beneficial to have a way to auth as part of this action as this action does the npm interaction for us -- conceptually keeps it all together.